xarray/tests/test_backends.py

```diff
@@ -992,6 +999,46 @@ def test_encoding_unlimited_dims(self):
             with pytest.warns(UserWarning):
                 ds.to_netcdf(tmp_file, engine='h5netcdf', unlimited_dims=['y'])
 
+class OpenMFDatasetTest(TestCase):
+    def validate_open_mfdataset_large_num_files(self, engine, nfiles=2000):
```

Comment:
But, this works for `nfiles=30`, not `nfiles=10` so we may be ok.  I'm going to include a commit with this and you both can rip it apart because it isn't 100% clear to me what to do here...